client/sites/overview/components/test/hosting-overview.tsx

```diff
@@ -0,0 +1,172 @@
+/**
+ * @jest-environment jsdom
+ */
+import { MigrationStatus } from '@automattic/data-stores';
+import React from 'react';
+import {
+	PURCHASES_SITE_FETCH,
+	SITE_REQUEST,
+	SITES_REQUEST,
+	SELECTED_SITE_SET,
+} from 'calypso/state/action-types';
+import {
+	renderWithProvider,
+	statefulRenderWithProvider,
+} from 'calypso/test-helpers/testing-library';
+import HostingOverview from '../hosting-overview';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { UnknownAction } from 'redux';
+
+type MockSiteOptions = {
+	mockSiteId: number;
+	is_wpcom_atomic?: boolean;
+	migrationStatus?: MigrationStatus;
+	migrationSticker?: string;
+};
+
+const buildMockSite = ( {
```

Comment:
It would be very nice if we started to build a centralized way to build factories, maybe using something like: 
https://www.npmjs.com/package/rosie